dali/test/python/test_pipeline.py

```diff
@@ -279,6 +279,51 @@ def iter_setup(self):
                 assert(np.sum(np.abs(t - t_pad[:,:,:3])) == 0)
                 assert(np.sum(np.abs(t_pad[:,:,3])) == 0)
 
+def test_cropmirrornormalize_multiple_inputs():
+    batch_size = 13
+    class HybridPipe(Pipeline):
+        def __init__(self, batch_size, num_threads=1, device_id=0, num_gpus=1, device="cpu"):
+            super(HybridPipe, self).__init__(batch_size,
+                                             num_threads,
+                                             device_id)
+            self.device = device
+            self.input = ops.CaffeReader(path = caffe_db_folder, shard_id = device_id, num_shards = num_gpus)
+            self.decode = ops.HostDecoder(device = "cpu", output_type = types.RGB)
+            self.decode2 = ops.HostDecoder(device = "cpu", output_type = types.RGB)
+            self.cmnp = ops.CropMirrorNormalize(device = device,
+                                                output_dtype = types.FLOAT,
+                                                output_layout = types.NHWC,
+                                                crop = (224, 224),
+                                                image_type = types.RGB,
+                                                mean = [128., 128., 128.],
+                                                std = [1., 1., 1.])
+
+        def define_graph(self):
+            inputs, labels = self.input(name="Reader")
+            images = self.decode(inputs)
+            images2 = self.decode2(inputs)
+
+            images_device  = images if self.device == "cpu" else images.gpu()
+            images2_device = images2 if self.device == "cpu" else images2.gpu()
+
+            output1, output2 = self.cmnp([images_device, images2_device])
+            output3 = self.cmnp([images_device])
+            output4 = self.cmnp([images2_device])
+            return (output1, output2, output3, output4)
+
+        def iter_setup(self):
+            pass
+
+    for device in ["cpu", "gpu"]:
```

Comment:
CPU and GPU data is not compared one to another?

Reply:
not in this test, but I believe that this is covered in the C++ tests